### ISSUE TITLE
Trying to figure out the phase factor when I put to determinants in maximum coincidence

### ISSUE THREAD

**Issue by pkozlows** [pkozlows/fci#6]
Say I have the pair det1={0,1,3,4,6,7} and det2={1,3,5,6,7,8}, so (det1.difference(det2), det2.difference(det1)) is something like ({0,4}, {5,8}]. How would I compute the face factor associated between these two determinants? I know that this is probably something that I need to figure out myself, but I was wondering if you could give any hints about what it would take to figure out the algorithm. I assume that I would somehow use second quantization. What is stumping me up currently is figuring out how I can compute a face factor with the determinants in maximum coincidence when my implementation for a determinant is, by nature, an unordered set in Python.

**Comment by pkozlows** [comment]
And if you are curious why my dictation text seems so accurate, but is written a little bit strangely, it is because I am using chatgpt to fix my raw dictation. ;)

**Comment by Walter-Feng** [comment]
I guess it is a hint I can provide: it is related to how you order the spin orbitals.

Say I have H 1s orbital, and normally I want to have it (1s up), (1s down) and this be the unique way of aligning the creation operators. And when another O 2px orbital come, I require two new spin orbitals inserted behind the H 1s spin orbitals, so on. 

After this unique order is defined, you will then need to solve the puzzle of creation/annihilation pairs, that come from, determinant on the right, on the left, and Hamiltonian. The only rule you can use is the commutation rules in second quantization, which is *local*, i.e. you can not teleport a creation operator to its destined fiancé and generate a integral - you will have to remove the obstacles for them by swapping around the NPCs and…. There’s the phase factor. 

You can argue, you propose a different way of sorting the orbitals, which will require a reorder of creation operators, which… you know lol

So basically the different definition of the order of creation operators will generally lead to different phase factor pattern, but that should not affect the eigenvalues you will solve for the Hamiltonian.

**Comment by Walter-Feng** [comment]
ChatGPT - it’s you again! Never imagined it can be worked like a charm here. I will be so interested how you figure these out!

**Comment by Walter-Feng** [comment]
By the way, it will be helpful if you can close the issues that you feel have been solved. That helps me quickly locate the issues you are having problem with.

If something new arises that are relevant with the closed issues, you can bring them up by reopening the issue, like what you did for the first issue. But keeping different topics separately in issues help both of us keep track of how the program is going on.

**Comment by pkozlows** [comment]
I definitely need to think more about how to implement second quantization into my code. Will there be any place for (det1.difference(det2), det2.difference(det1)) in my code or will I just be using creation and annihilation operators acting on a vaccum state Too construct and modify the determinants? Or maybe, the questions I'm asking are too confusing for you in which case I need to think that more about second quantization before I ask more questions.

**Comment by Walter-Feng** [comment]
I believe you will eventually need to make a difference between two determinants, as an excitation pair is inevitable when writing FCI Hamiltonian elements. 

Maybe no need to treat the creation/annihilation operators too seriously, as you will get used to it. Just as a reminder, when you stuff an electron into an orbital, it is equivalent to adding a creation operator to your state. So when you write determinants, you also generated equivalent string of creation operators.

By the way, the final shape of the program that you will hand in to our advisor may not include an exact difference of two determinants, as it uses a different algorithm. Nevertheless, we are still far away from that - just focus on generating the FCI Hamiltonian first. We will be there.
